### PR TITLE
Fix error management

### PR DESCRIPTION
**Pull request by Harshith46**
# Description

Fix error management in request function, previously we we passing the error object as the response status was 200

Fixes: https://github.com/quintype/ace-planning/issues/507

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test Case A
- [ ] Test Case B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: collection.spec.js [hunk 1]
```diff
@@ -1,143 +1,152 @@
-const {Collection} = require('./index');
+const { Collection } = require("./index");
+const _ = require("lodash");
```
Comment (Deokumarjnu) [inline collection.spec.js:2]: Can you only import required function from ladash

Comment (Harshith46) [reply collection.spec.js:2]: sure

File: collection.spec.js [hunk 2]
```diff
@@ -1,143 +1,152 @@
...
+        qs: params
+      });
+    } catch (e) {
+      return catch404(e, null);
+    }
+  }
+} = {}) {
+  return { getCollectionBySlug };
+}
+
```
Comment (ags1773) [inline collection.spec.js:80]: Instead, you can reuse the `getClient` method above by passing it `getCollectionBySlug` function as a parameter

Comment (ags1773) [reply collection.spec.js:80]:
```js
const clientThatCallsCollThatDoesntExist = getClient({
  getCollectionBySlug: () => {
    return Promise.resolve({response: { status: 404 }})
  }
})
```

**Review by ags1773 — APPROVED** [review]

**Pull request merged**